lib/msal-core/src/UserAgentApplication.ts

```diff
@@ -388,201 +378,529 @@ export class UserAgentApplication {
+      }
+
+      // Close the popup window
+      if (popUpWindow) {
+        popUpWindow.close();
+      }
+    }).catch((err) => {
+      // All catch - when is this executed? Possibly when error is thrown, but not if previous function rejects instead of throwing
+      this._logger.warning("could not resolve endpoints");
+      reject(err);
+    });
   }
 
   /**
-   * Used to send the user to the redirect_uri after authentication is complete. The user"s bearer token is attached to the URI fragment as an id_token/access_token field.
-   * This function also closes the popup window after redirection.
-   * @hidden
-   * @ignore
-   */
-  private openWindow(urlNavigate: string, title: string, interval: number, instance: this, resolve?: Function, reject?: Function): Window {
-    var popupWindow = this.openPopup(urlNavigate, title, Constants.popUpWidth, Constants.popUpHeight);
-    if (popupWindow == null) {
-      instance._loginInProgress = false;
+   * Used to acquire an access token for a new user using interactive authentication via a popup Window.
+   * To request an id_token, pass the clientId as the only scope in the scopes array.
+   * @param {Array<string>} scopes - Permissions you want included in the access token. Not all scopes are  guaranteed to be included in the access token. Scopes like "openid" and "profile" are sent with every request.
+   * @param {string} authority - A URL indicating a directory that MSAL can use to obtain tokens.
+   * - In Azure AD, it is of the form https://&lt;tenant&gt;/&lt;tenant&gt;, where &lt;tenant&gt; is the directory host (e.g. https://login.microsoftonline.com) and &lt;tenant&gt; is a identifier within the directory itself (e.g. a domain associated to the tenant, such as contoso.onmicrosoft.com, or the GUID representing the TenantID property of the directory)
+   * - In Azure B2C, it is of the form https://&lt;instance&gt;/tfp/&lt;tenant&gt;/<policyName>/
+   * - Default value is: "https://login.microsoftonline.com/common".
+   * @param {User} user - The user for which the scopes are requested.The default user is the logged in user.
+   * @param {string} extraQueryParameters - Key-value pairs to pass to the STS during the  authentication flow.
+   * @returns {Promise.<string>} - A Promise that is fulfilled when this function has completed, or rejected if an error was raised. Returns the token or error.
+   */
+  // TODO: params to accept AuthRequest object instead
+  acquireTokenPopup(scopes: Array<string>): Promise<string>;
+  acquireTokenPopup(scopes: Array<string>, authority: string): Promise<string>;
+  acquireTokenPopup(scopes: Array<string>, authority: string, user: User): Promise<string>;
+  acquireTokenPopup(scopes: Array<string>, authority: string, user: User, extraQueryParameters: string): Promise<string>;
+  acquireTokenPopup(scopes: Array<string>, authority?: string, user?: User, extraQueryParameters?: string): Promise<string> {
```

Comment:
same question here about unnecessary signature

Reply:
This will be a single API surface after Request. This is legacy with various overloads supported and we are trying to move away from this model with the upcoming PRs.